File: backend/app/schemas/clinic.py
from pydantic import BaseModel


class ErrorPatternOut(BaseModel):
    id: int
    pattern_type: str
    title: str
    description: str
    severity: str
    evidence_json: dict | None = None
    diagnosis_json: dict | None = None
    status: str
    created_at: str


class DiagnosisReport(BaseModel):
    patterns: list[ErrorPatternOut]
    summary: str
    total_errors_analyzed: int


class TreatmentPlanOut(BaseModel):
    id: int
    pattern_id: int
    exercises_json: dict | None = None
    total_exercises: int
    completed_exercises: int
    status: str


class TreatmentExerciseSubmit(BaseModel):
    plan_id: int
    exercise_index: int
    answer: str


class TreatmentExerciseResult(BaseModel):
    is_correct: bool
    correct_answer: str
    explanation: str
    progress: int
    total: int
    plan_completed: bool
    xp: dict | None = None

File: frontend/src/stores/clinic.ts
import { create } from "zustand";
import { api } from "@/lib/api";

interface ErrorPattern {
  id: number;
  pattern_type: string;
  title: string;
  description: string;
  severity: string;
  evidence_json: Record<string, unknown> | null;
  diagnosis_json: Record<string, unknown> | null;
  status: string;
  created_at: string;
}

interface TreatmentPlan {
  id: number;
  pattern_id: number;
  exercises_json: { exercises: { type: string; question: string; options?: string[]; answer: string; explanation: string }[] } | null;
  total_exercises: number;
  completed_exercises: number;
  status: string;
}

interface ClinicState {
  patterns: ErrorPattern[];
  currentPlan: TreatmentPlan | null;
  diagnosing: boolean;
  summary: string;
  diagnose: () => Promise<void>;
  fetchPatterns: () => Promise<void>;
  startTreatment: (patternId: number) => Promise<void>;
  submitExercise: (planId: number, index: number, answer: string) => Promise<Record<string, unknown>>;
}

export const useClinicStore = create<ClinicState>((set, get) => ({
  patterns: [],
  currentPlan: null,
  diagnosing: false,
  summary: "",

  diagnose: async () => {
    set({ diagnosing: true });
    try {
      const data = await api.post<{ patterns: ErrorPattern[]; summary: string }>("/clinic/diagnose", {});
      set({ patterns: data.patterns, summary: data.summary, diagnosing: false });
    } catch {
      set({ diagnosing: false });
    }
  },

  fetchPatterns: async () => {
    try {
      const data = await api.get<ErrorPattern[]>("/clinic/patterns");
      set({ patterns: data });
    } catch { /* ignore */ }
  },

  startTreatment: async (patternId: number) => {
    try {
      const data = await api.post<TreatmentPlan>(`/clinic/treat/${patternId}`, {});
      set({ currentPlan: data });
    } catch { /* ignore */ }
  },

  submitExercise: async (planId: number, index: number, answer: string) => {
    const res = await api.post<Record<string, unknown>>("/clinic/exercise", {
      plan_id: planId, exercise_index: index, answer,
    });
    if (res.plan_completed) {
      set({ currentPlan: null });
      get().fetchPatterns();
    }
    return res;
  },
}));
